streaming/src/main/scala/org/apache/spark/streaming/scheduler/ReceiverTracker.scala

```diff
@@ -431,7 +450,8 @@ class ReceiverTracker(ssc: StreamingContext, skipReceiverLaunch: Boolean = false
           receiver.preferredLocation,
           receiverTrackingInfos,
           getExecutors)
-        updateReceiverScheduledExecutors(receiver.streamId, scheduledExecutors)
```

Comment:
Why not update the scheduled info in case of `rescheduleReceiver`? Why have these two different code paths / policies in two cases.